### PR TITLE
__Invert__  instance method - Numpy's ndarray class

### LINKED ISSUE #18284: __invert__

### PR DESCRIPTION
**Pull request by AsmaaMHadir**
Close #18284

### CODE REVIEW

**Comment by AsmaaMHadir** [comment]
ivy-gardener

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 1]
```diff
@@ -53,15 +53,15 @@ def shape(self):
     def dtype(self):
         return self.ivy_array.dtype
 
-    @property
-    def ndim(self):
-        return len(self.shape)
-
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:59]: unnecessary change

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 2]
```diff
@@ -92,29 +92,6 @@ def test_numpy_ndarray_property_ndim(
     ivy.utils.assertions.check_equal(x.ndim, data[0].ndim, as_array=False)
 
 
-@given(
```
Comment (Sam-Armstrong) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:95]: why have you removed this test?

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
@AsmaaMHadir thanks for making the PR! Please could you revert moving the ndim property and removing test_numpy_ndarray_property_T? Any changes in the pull request should be directly related to the function you're working on. Thanks 🙂

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py (hunk 2 above)
Comment (AsmaaMHadir) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:95]: Hey Sam -  Thanks a lot for your feedback, I'm not sure how I came to remove it accidentally! rolling it back :)

File: ivy/functional/frontends/numpy/ndarray/ndarray.py (hunk 1 above)
Comment (AsmaaMHadir) [reply ivy/functional/frontends/numpy/ndarray/ndarray.py:59]: Same thing here. I'm not sure how I changed the ndim() property. applying these changes right away :)

**Comment by AsmaaMHadir** [comment]
Hey @Sam-Armstrong - I am not sure how to debug the failing checks. Could you please guide me in replicating/ resolving them?

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 3]
```diff
@@ -3145,3 +3145,37 @@ def test_numpy_instance_tostring__(
...
     )
+
+
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="numpy.array",
+    method_name="__invert__",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes(kind="integer"),
+        num_arrays=2,
```
Comment (Sam-Armstrong) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:3156]:
```suggestion
        num_arrays=1,
```

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 4]
```diff
@@ -500,6 +500,9 @@ def __ixor__(self, value, /):
     def __imod__(self, value, /):
         return np_frontend.mod(self, value, out=self)
 
+    def __invert__(self, out=None, /):
+        return ivy.bitwise_invert(self.ivy_array, out=out)
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:504]:
```suggestion
    def __invert__(self, /):
        return ivy.bitwise_invert(self.ivy_array)
```

Comment (Sam-Armstrong) [reply ivy/functional/frontends/numpy/ndarray/ndarray.py:504]: We don't need any out argument here - there is no out arg in the numpy function https://numpy.org/doc/stable/reference/generated/numpy.ndarray.__invert__.html

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
Don't worry about the checks failing, that is caused by other parts of ivy. As long as your tests pass then we are good 😊 I've left a couple of suggested changes, please could you address these, then we should be good to merge. Thanks!

**Review by Sam-Armstrong — APPROVED** [review]
lgtm, thanks @AsmaaMHadir!
